code/modules/mob/living/carbon/xenomorph/Castes/Sentinel.dm

```diff
@@ -204,27 +204,37 @@
 
 	face_atom(H)
 	animation_attack_on(H)
-	H.reagents.add_reagent("xeno_toxin", NEUROTOXIN_STING_AMOUNT_INITIAL) //15 units transferred initially.
+	H.reagents.add_reagent("xeno_toxin", SENTINEL_STING_AMOUNT_INITIAL) //15 units transferred initially.
 	to_chat(H, "<span class='danger'>You feel a tiny prick.</span>")
 	to_chat(src, "<span class='xenowarning'>Your stinger injects your victim with neurotoxin!</span>")
 	playsound(H, 'sound/effects/spray3.ogg', 15, 1)
 	playsound(H, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
-	if(do_after(src, NEUROTOXIN_STING_INJECT_DELAY, TRUE, 5, BUSY_ICON_HOSTILE)) //First follow up injection
-		if(!neurotoxin_recurring_injection(H))
-			return
-		to_chat(src, "<span class='xenowarning'>Your stinger continues to inject neurotoxin!</span>")
-		if(do_after(src, NEUROTOXIN_STING_INJECT_DELAY, TRUE, 5, BUSY_ICON_HOSTILE)) //Second follow up injection
-			if(!neurotoxin_recurring_injection(H))
-				return
-			to_chat(src, "<span class='xenowarning'>Your stinger retracts as it finishes discharging the neurotoxin.</span>")
+	overdose_check(H)
+
+	addtimer(CALLBACK(src, .sentinel_sting_cooldown), SENTINEL_STING_COOLDOWN)
 
-	addtimer(CALLBACK(src, .neurotoxin_sting_cooldown), NEUROTOXIN_STING_COOLDOWN)
+	sentinel_recurring_injection(H)
 
-/mob/living/carbon/Xenomorph/Sentinel/proc/neurotoxin_sting_cooldown()
+/mob/living/carbon/Xenomorph/Sentinel/proc/sentinel_sting_cooldown()
 	playsound(loc, 'sound/voice/alien_drool1.ogg', 50, 1)
-	to_chat(src, "<span class='xenodanger'>You feel your neurotoxin glands refill. You can use your neurotoxin sting again.</span>")
+	to_chat(src, "<span class='xenodanger'>You feel your neurotoxin glands refill. You can use your Neurotoxin Sting again.</span>")
 	update_action_button_icons()
 
+/mob/living/carbon/Xenomorph/Sentinel/proc/sentinel_recurring_injection(mob/living/H, count = 2)
+	//set waitfor = FALSE
+	while(count)
+		if(!Adjacent(H) || stagger)
```

Comment:
Same issue with `drone_recurring_injection()`. In fact both procs are similar enough that there's no need to make two different ones. Merge them into one, please.